enterprise/cypher/spec-suite-tools/src/test/scala/cypher/features/ScenarioTestHelper.scala

```diff
@@ -0,0 +1,102 @@
+ */
+package cypher.features
+
+import java.io.FileNotFoundException
+import java.net.URI
+import java.nio.charset.StandardCharsets
+import java.util
+
+import org.junit.jupiter.api.DynamicTest
+import org.junit.jupiter.api.function.Executable
+import org.opencypher.tools.tck.api.Scenario
+
+import scala.collection.JavaConverters._
+import scala.io.Source
+import scala.util.{Failure, Success, Try}
+
+object ScenarioTestHelper {
+  def createTests(scenarios: Seq[Scenario],
+                  config: TestConfig,
+                  debugOutput: Boolean = false): util.Collection[DynamicTest] = {
+    val blacklist = config.blacklist.map(parseBlacklist).getOrElse(Set.empty[BlacklistEntry])
+    val (expectFail, expectPass) = scenarios.partition { s => blacklist.exists(_.isBlacklisted(s)) }
+    if (debugOutput) {
+      val unusedBlacklistEntries = blacklist.filterNot(b => expectFail.exists(s => b.isBlacklisted(s)))
+      if (unusedBlacklistEntries.nonEmpty) {
+        println("The following entries of the blacklist were not found in the blacklist: \n"
+          + unusedBlacklistEntries.mkString("\n"))
+      }
+    }
+
+    val expectFailTests: Seq[DynamicTest] = expectFail.map { scenario =>
+      val name = scenario.toString()
+      val executable = new Executable {
+        override def execute(): Unit = {
+          Try {
+            scenario(Neo4jAdapter(config.executionPrefix)).execute()
+          } match {
+            case Success(_) => throw new IllegalStateException("Unexpectedly succeeded in the following blacklisted scenario:\n" + name)
+            case Failure(e) => println("Failed as expected with\n  " + e.getMessage)
+          }
+        }
+      }
+      DynamicTest.dynamicTest("Blacklisted Test: " + name, executable)
+    }
+
+    val expectPassTests: Seq[DynamicTest] = expectPass.map { scenario =>
+      val name = scenario.toString()
+      val executable = scenario(Neo4jAdapter(config.executionPrefix))
+      DynamicTest.dynamicTest(name, executable)
+    }
+    (expectPassTests ++ expectFailTests).asJavaCollection
+  }
+
+  def parseBlacklist(blacklistFile: String): Set[BlacklistEntry] = {
+    def validate(scenarioName: String): Unit = {
+      if (scenarioName.head.isWhitespace || scenarioName.last.isWhitespace) {
+        throw new Exception(s"Invalid whitespace in scenario name $scenarioName from file $blacklistFile")
+      }
+    }
+
+    val uri = new URI("/blacklists/" + blacklistFile)
+    val url = getClass.getResource(uri.getPath)
+    if (url == null) throw new FileNotFoundException(s"Blacklist file not found at: $blacklistFile")
+    val lines = Source.fromFile(url.getPath, StandardCharsets.UTF_8.name()).getLines()
+    val scenarios = lines.filterNot(line => line.startsWith("//") || line.isEmpty).toSet // comments in blacklist are being ignored
+    scenarios.foreach(validate)
+    scenarios.map(BlacklistEntry(_))
+  }
+
+  def printComputedBlacklist(scenarios: Seq[Scenario],
```

Comment:
We should put a comment here, so that the method does not get deleted by accident, even though it is unused by default.